src/ResultPaginator.php

```diff
@@ -29,6 +29,10 @@ class ResultPaginator implements \Iterator
     /** @var int Number of operations/requests performed. */
     private $requestCount = 0;
 
+    /** @var array Operations whose last page of results
+     *             returns the previous page's NextToken. */
+    private $lastPageHasPreviousToken = ['GetLogEvents'];
```

Comment:
This variable is named like a bool, should be something like $lastPageHasPreviousTokenOperations, but more succinct if you can come up with something